### PR TITLE
Create course controller with get all endpoint and tests 

### LINKED ISSUE #34: Subfeature: Add Get All endpoint for Courses table
Part of Epic #6 
Part of Feature #11 

Adds controller class with the get /all endpoint to Courses.

get all should be viewable from the swagger and have relevant test for permissions.

### PR DESCRIPTION
**Pull request by r0ckwav3**
## Overview
Add a course controller with the ability to get all as well as tests for get all.
Dokku Deployment (database is still empty): https://happycows-dev-r0ckwav3.dokku-10.cs.ucsb.edu/

## Validation 

1. Go to the dokku deployment's swagger
2. Verify that the get all endpoint exists

Or

1. Download the branch locally
2. Run `mvn test`

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #34

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CourseController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@Slf4j
+@Tag(name = "Course")
+@RequestMapping("/api/course")
+@RestController
+public class CourseController extends ApiController {
+    @Autowired
+    private CourseRepository courseRepository;
+
+    /**
+     * THis method returns a list of all courses.
```
Comment (rranu) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CourseController.java:21]: Small capitalization change in comment
```suggestion
     * This method returns a list of all courses.
```

**Review by rranu — DISMISSED** [review]
LGTM! Made one suggestion to fix capitalization on a comment but that doesn't pertain to the actual code so this review will approve with feedback.

**Review by rranu — APPROVED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CourseControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+@Import(TestConfig.class)
+public class CourseControllerTests extends ControllerTestCase {
+    @MockBean
+    CourseRepository courseRepository;
+
+    @MockBean
+    UserRepository userRepository;
+
+    // @Autowired
+    // private ObjectMapper objectMapper;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/CourseControllerTests.java:33]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — DISMISSED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CourseControllerTests.java (hunk 2 above)
Comment (r0ckwav3) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/CourseControllerTests.java]: yep good catch!

**Review by rranu — APPROVED** [review]

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**